### PR TITLE
MAINT Update the packages.json info to include emscripten version and pyodide version

### PR DESCRIPTION
**Pull request by hoodmane**
Split from #2645

### CODE REVIEW

File: pyodide-build/pyodide_build/buildall.py [hunk 1]
```diff
@@ -457,9 +457,16 @@ def generate_packages_json(
     output_dir: Path, pkg_map: dict[str, BasePackage]
 ) -> dict[str, dict[str, Any]]:
     """Generate the package.json file"""
+
+    import sys
+
+    sys.path.append(str(common.get_pyodide_root() / "src/py"))
+    from pyodide import __version__
```
Comment (rth) [inline pyodide-build/pyodide_build/buildall.py:464]: I'm not sure that this package would always be importable on the host in the future, but we can deal with that issue when it happens.

**Review by rth — APPROVED** [review]
Thanks!

File: pyodide-build/pyodide_build/buildall.py [hunk 2]
```diff
@@ -457,9 +457,16 @@ def generate_packages_json(
     output_dir: Path, pkg_map: dict[str, BasePackage]
 ) -> dict[str, dict[str, Any]]:
     """Generate the package.json file"""
```
Comment (ryanking13) [inline pyodide-build/pyodide_build/buildall.py:459]: For example, even maintainers often make a typo just like here.

**Review by ryanking13 — COMMENTED** [review]
Thanks @hoodmane! Probably we should consider renaming `packages.json` as it is starting to contain more than just package information and sometimes it is confusing as its name is similar to `package.json` of Node.js.

**Review by ryanking13 — APPROVED** [review]
Thanks, @hoodmane! There is a failing test, otherwise LGTM.

**Comment by hoodmane** [comment]
Okay I'll open a followup to change `packages.json` to `repodata.json`.

**Pull request merged**